### ISSUE TITLE
SES check fails in some regions

### ISSUE THREAD

**Issue by jwu2** [jantman/awslimitchecker#464]
### Version

Docker, 8.0.2

### Installation Method

Docker

### Actual Output

Four regions have this issue:
Seoul:ap-northeast-2, Tokyo:ap-northeast-1, Singapore:ap-southeast-1, Ohio:us-east-2

```
docker run -e AWS_DEFAULT_REGION=$AWS_DEFAULT_REGION -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY -e AWS_SESSION_TOKEN=$AWS_SESSION_TOKEN jantman/awslimitchecker -S SES -r us-east-2 -vv

2020-05-01 01:47:40,836 [CRITICAL subprocess.py:195 - pip.subprocessor.call_subprocess() ] Error [Errno 2] No such file or directory: 'git' while executing command git config --get-regexp 'remote\..*\.url'
awslimitchecker 8.0.2 is AGPL-licensed free software; all users have a right to the full source code of this version. See <https://github.com/jantman/awslimitchecker>
2020-05-01 01:47:40,841 [DEBUG connectionpool.py:955 - urllib3.connectionpool._new_conn() ] Starting new HTTPS connection (1): pypi.org:443
2020-05-01 01:47:40,972 [DEBUG connectionpool.py:428 - urllib3.connectionpool._make_request() ] https://pypi.org:443 "GET /pypi/awslimitchecker/json HTTP/1.1" 200 64169
2020-05-01 01:47:41,038 [DEBUG checker.py:285 - awslimitchecker.checker._boto_conn_kwargs() ] Connecting to region us-east-2
2020-05-01 01:47:41,039 [DEBUG cloudtrail.py:125 - awslimitchecker.services.cloudtrail.get_limits() ] Gathering CloudTrail's limits from AWS
2020-05-01 01:47:41,082 [INFO connectable.py:119 - awslimitchecker.connectable.connect() ] Connected to dynamodb in region us-east-2
2020-05-01 01:47:41,127 [INFO connectable.py:119 - awslimitchecker.connectable.connect() ] Connected to ec2 in region us-east-2
2020-05-01 01:47:41,128 [DEBUG lambdafunc.py:82 - awslimitchecker.services.lambdafunc.get_limits() ] Getting limits for Lambda
2020-05-01 01:47:41,128 [DEBUG cloudtrail.py:125 - awslimitchecker.services.cloudtrail.get_limits() ] Gathering CloudTrail's limits from AWS
2020-05-01 01:47:41,128 [DEBUG lambdafunc.py:82 - awslimitchecker.services.lambdafunc.get_limits() ] Getting limits for Lambda
2020-05-01 01:47:41,134 [INFO connectable.py:119 - awslimitchecker.connectable.connect() ] Connected to support in region us-east-1
2020-05-01 01:47:41,134 [INFO trustedadvisor.py:166 - awslimitchecker.trustedadvisor._poll() ] Beginning TrustedAdvisor poll
2020-05-01 01:47:41,134 [DEBUG trustedadvisor.py:223 - awslimitchecker.trustedadvisor._get_limit_check_id() ] Querying Trusted Advisor checks
2020-05-01 01:47:41,136 [DEBUG connectionpool.py:955 - urllib3.connectionpool._new_conn() ] Starting new HTTPS connection (1): support.us-east-1.amazonaws.com:443
2020-05-01 01:47:42,674 [DEBUG connectionpool.py:428 - urllib3.connectionpool._make_request() ] https://support.us-east-1.amazonaws.com:443 "POST / HTTP/1.1" 200 169892
2020-05-01 01:47:45,513 [DEBUG trustedadvisor.py:248 - awslimitchecker.trustedadvisor._get_limit_check_id() ] Unable to find check with category 'performance' and name 'Service Limits'.
2020-05-01 01:47:45,513 [CRITICAL trustedadvisor.py:172 - awslimitchecker.trustedadvisor._poll() ] Unable to find 'Service Limits' Trusted Advisor check; not using Trusted Advisor data.
2020-05-01 01:47:45,513 [DEBUG trustedadvisor.py:412 - awslimitchecker.trustedadvisor._update_services() ] Updating TA limits on all services
2020-05-01 01:47:45,513 [INFO trustedadvisor.py:432 - awslimitchecker.trustedadvisor._update_services() ] Done updating TA limits on all services
2020-05-01 01:47:45,522 [INFO connectable.py:119 - awslimitchecker.connectable.connect() ] Connected to ses in region us-east-2
2020-05-01 01:47:45,523 [DEBUG connectionpool.py:955 - urllib3.connectionpool._new_conn() ] Starting new HTTPS connection (1): email.us-east-2.amazonaws.com:443
2020-05-01 01:50:46,115 [DEBUG connectionpool.py:955 - urllib3.connectionpool._new_conn() ] Starting new HTTPS connection (2): email.us-east-2.amazonaws.com:443
2020-05-01 01:53:46,830 [DEBUG connectionpool.py:955 - urllib3.connectionpool._new_conn() ] Starting new HTTPS connection (3): email.us-east-2.amazonaws.com:443
2020-05-01 01:56:49,690 [DEBUG connectionpool.py:955 - urllib3.connectionpool._new_conn() ] Starting new HTTPS connection (4): email.us-east-2.amazonaws.com:443
2020-05-01 01:59:55,860 [DEBUG connectionpool.py:955 - urllib3.connectionpool._new_conn() ] Starting new HTTPS connection (5): email.us-east-2.amazonaws.com:443
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 156, in _new_conn
    conn = connection.create_connection(
  File "/usr/local/lib/python3.8/site-packages/urllib3/util/connection.py", line 84, in create_connection
    raise err
  File "/usr/local/lib/python3.8/site-packages/urllib3/util/connection.py", line 74, in create_connection
    sock.connect(sa)
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/botocore/httpsession.py", line 254, in send
    urllib_response = conn.urlopen(
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 719, in urlopen
    retries = retries.increment(
  File "/usr/local/lib/python3.8/site-packages/urllib3/util/retry.py", line 376, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/local/lib/python3.8/site-packages/urllib3/packages/six.py", line 735, in reraise
    raise value
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 665, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 376, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 994, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 300, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.8/site-packages/urllib3/connection.py", line 161, in _new_conn
    raise ConnectTimeoutError(
urllib3.exceptions.ConnectTimeoutError: (<botocore.awsrequest.AWSHTTPSConnection object at 0x7ffa4a11a280>, 'Connection to email.us-east-2.amazonaws.com timed out. (connect timeout=60)')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/awslimitchecker", line 11, in <module>
    load_entry_point('awslimitchecker', 'console_scripts', 'awslimitchecker')()
  File "/awslimitchecker/awslimitchecker/runner.py", line 548, in console_entry_point
    r.console_entry_point()
  File "/awslimitchecker/awslimitchecker/runner.py", line 520, in console_entry_point
    res, problems, problem_str = self.check_thresholds(metrics)
  File "/awslimitchecker/awslimitchecker/runner.py", line 332, in check_thresholds
    problems = self.checker.check_thresholds(
  File "/awslimitchecker/awslimitchecker/checker.py", line 631, in check_thresholds
    cls._update_limits_from_api()
  File "/awslimitchecker/awslimitchecker/services/ses.py", line 111, in _update_limits_from_api
    resp = self.conn.get_send_quota()
  File "/usr/local/lib/python3.8/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.8/site-packages/botocore/client.py", line 612, in _make_api_call
    http, parsed_response = self._make_request(
  File "/usr/local/lib/python3.8/site-packages/botocore/client.py", line 632, in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
  File "/usr/local/lib/python3.8/site-packages/botocore/endpoint.py", line 102, in make_request
    return self._send_request(request_dict, operation_model)
  File "/usr/local/lib/python3.8/site-packages/botocore/endpoint.py", line 136, in _send_request
    while self._needs_retry(attempts, operation_model, request_dict,
  File "/usr/local/lib/python3.8/site-packages/botocore/endpoint.py", line 228, in _needs_retry
    responses = self._event_emitter.emit(
  File "/usr/local/lib/python3.8/site-packages/botocore/hooks.py", line 356, in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/botocore/hooks.py", line 228, in emit
    return self._emit(event_name, kwargs)
  File "/usr/local/lib/python3.8/site-packages/botocore/hooks.py", line 211, in _emit
    response = handler(**kwargs)
  File "/usr/local/lib/python3.8/site-packages/botocore/retryhandler.py", line 183, in __call__
    if self._checker(attempts, response, caught_exception):
  File "/usr/local/lib/python3.8/site-packages/botocore/retryhandler.py", line 250, in __call__
    should_retry = self._should_retry(attempt_number, response,
  File "/usr/local/lib/python3.8/site-packages/botocore/retryhandler.py", line 277, in _should_retry
    return self._checker(attempt_number, response, caught_exception)
  File "/usr/local/lib/python3.8/site-packages/botocore/retryhandler.py", line 316, in __call__
    checker_response = checker(attempt_number, response,
  File "/usr/local/lib/python3.8/site-packages/botocore/retryhandler.py", line 222, in __call__
    return self._check_caught_exception(
  File "/usr/local/lib/python3.8/site-packages/botocore/retryhandler.py", line 359, in _check_caught_exception
    raise caught_exception
  File "/usr/local/lib/python3.8/site-packages/botocore/endpoint.py", line 200, in _do_get_response
    http_response = self._send(request)
  File "/usr/local/lib/python3.8/site-packages/botocore/endpoint.py", line 244, in _send
    return self.http_session.send(request)
  File "/usr/local/lib/python3.8/site-packages/botocore/httpsession.py", line 287, in send
    raise ConnectTimeoutError(endpoint_url=request.url, error=e)
botocore.exceptions.ConnectTimeoutError: Connect timeout on endpoint URL: "https://email.us-east-2.amazonaws.com/"
```

### Expected Output

Pass without error.

### Testing Assistance

I will assist in testing pre-release
code for the feature.

**Comment by jantman** [comment]
@jwu2 Thanks for reporting this! It looks like they may be preparing to enable SES in those regions! We [currently have code](https://github.com/jantman/awslimitchecker/blob/2bb3ae7a8f257506022b61896507b937a6c0ec88/awslimitchecker/services/ses.py#L65-L75) to handle EndpointConnectionErrors and 503s from SES, which previously handled the regions that don't have it... but it appears that we (or at least you) are now getting Connection Timeouts from those regions.

For an immediate fix, I'd advise no longer checking SES in the problematic regions (this can be done with the ``--skip-service SES`` command line argument).

As is generally the case with issues like this, I'd like to wait 1-3 days to see if anyone can confirm this issue (or if I can confirm it myself). I'll try to get a fix in the works within the next few days, with a target of releasing it early next week.

Thanks so much for the feedback!

**Comment by rosenta** [comment]
Hello @jantman we'v been using awslimitchecker CLI for workflow orchestration in our organization and I've found that, it failed with same `botocore.exceptions.ConnectTimeoutError"` at us-east-1 region. inorder to resolve that, we can use retry to handle this issue but would be great if it can possibly be integrated. and seriously thanks for creating this module, 
implementing AWS quota usage at this high level would be really beyond of our eye-sight for us.

**Comment by JTarasovic** [comment]
I'm seeing the same thing. Already did `--skip-service SES` which, of course, worked like a charm. :+1:

**Comment by jantman** [comment]
Ok, I'm getting the same thing now from SES in us-west-1. It appears to be persistent for my account for the last 5 days.

**Comment by jantman** [comment]
The fix for this has been released in [8.1.0](https://github.com/jantman/awslimitchecker/releases/tag/8.1.0) which is now live [on PyPI](https://pypi.org/project/awslimitchecker/8.1.0/) and live [on Docker Hub](https://hub.docker.com/r/jantman/awslimitchecker).

Thank you so much!!
